packages/block-library/src/gallery/use-image-sizes.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import { get, reduce, map, filter, some } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+export default function useImageSizes( images, isSelected, getSettings ) {
+	return useMemo( () => getImageSizing(), [ images, isSelected ] );
+
+	function getImageSizing() {
+		if ( ! images || images.length === 0 ) {
+			return [];
+		}
+		const { imageSizes } = getSettings();
+		let resizedImages = {};
+
+		if ( isSelected ) {
+			resizedImages = reduce(
+				images,
+				( currentResizedImages, img ) => {
+					if ( ! img.id ) {
+						return currentResizedImages;
+					}
+					const image = img;
```

Comment:
I think the creation of this variable is unnecessary. `img` could just be used directly and/or renamed to `image`.

Reply:
hmm, not sure why that was there - fixed up now.